File: s3PythonSDKExample/sample/PolicySerialTesting.py
from botocore.exceptions import ClientError
from client import client
# Use http://awspolicygen.s3.amazonaws.com/policygen.html to generate policy String
# test 1.Put bucket
#      2.Put policy
#      3.Get policy
#      4.Delete policy


def main(arg):
    try:
        client.create_bucket(
            CreateBucketConfiguration={'LocationConstraint': 'ap-southeast-1'},
            Bucket=arg[0]
        )
        #print("Put Deny Public-read policy to bucket")
        client.put_bucket_policy(
            Bucket=arg[0],
            Policy='{"Version":"2012-10-17","Statement":[{"Sid":"DenyPublicREAD","Effect":"Deny","Principal":{"AWS":"*"},"Action":"s3:GetObject","Resource":"arn:aws:s3:::'+arg[0]+'/*"}]}',
        )
        #print("Get Policy:")
        response = client.get_bucket_policy(
            Bucket=arg[0]
        )
        # print(repr(response['Policy']))
        client.delete_bucket_policy(
            Bucket=arg[0]
        )
        #print("\n Clean up..")
        client.delete_bucket(
            Bucket=arg[0],
        )
        print("Policy Serial Test Done !\n")
    except ClientError as e:
        print(e)

File: s3RubySDKExample/sample/policy_serial_testing.rb
def policy_serial_testing(s3, buckets)
  resp = s3.create_bucket({
    acl: "public-read", # accepts private, public-read, public-read-write, authenticated-read
    bucket: buckets[0], # required
  })

  myPolicy = '{
  "Id": "Policy1436779373666",
  "Version": "2012-10-17",
  "Statement": [
    {
      "Sid": "Stmt1436779368306",
      "Action": [
        "s3:GetObject"
      ],
      "Effect": "Deny",
      "Resource": "arn:aws:s3:::' + buckets[0] + '/*",
      "Principal": "*"
    }
  ]
  }'

  resp = s3.put_bucket_policy({
    bucket: buckets[0], # required
    #    content_md5: "ContentMD5",
    policy: myPolicy, # required
  })

  resp = s3.get_bucket_policy({
    bucket: buckets[0], # required
  })

  puts resp.policy

  resp = s3.delete_bucket_policy({
    bucket: buckets[0], # required
  })

  # clean all buckets
  cleanup(s3, buckets)
end
